cue.c

```diff
@@ -481,7 +481,17 @@ struct cue_sheet *cue_from_file(const char *file)
 	char *buf = mmap_file(file, &size);
 	if (size == -1)
 		return NULL;
-	struct cue_sheet *rv = cue_parse(buf, size);
+	struct cue_sheet *rv;
+
+	// Check for UTF-8 BOM, and skip ahead if found
+	if (size >= 3 && memcmp(buf, "\xEF\xBB\xBF", 3) == 0) {
+		ssize_t postbom_size = size - 3;
+		char *postbom_buf = cue_strdup(&buf[3], postbom_size);
+		rv = cue_parse(postbom_buf, postbom_size);
```

Comment:
Why `cue_strdup()`? Isn't

```C
rv = cue_parse(buf + 3, size - 3)
```
enough?

Reply:
I'm a dolt, way overthinking that because my recollection of c is a lot more convoluted than it really is. Thanks for your patience with this.